src/main/java/io/vlingo/wire/fdx/bidirectional/SecureClientRequestResponseChannel.java

```diff
@@ -209,7 +202,7 @@ public void onSuccess() {
 
     @Override
     public void onInput(final ByteBuffer decrypted) {
-      final ConsumerByteBuffer buffer = readBufferPool.accessFor("ssl-read");
+      final ConsumerByteBuffer buffer = readBufferPool.acquire();
       consumer.consume(buffer.put(decrypted).flip());
```

Comment:
Can `put` be avoided?

Reply:
Do you mean that the copy is unnecessary? Otherwise, what is the problem you see and how would you do it otherwise? BTW, I did not implement the RSocket support so I am a bit out of context here. Thanks for explaining a bit more.

Reply:
I am concerned about the allocation overhead of copying potentially large buffers. There is a documented case here: https://github.com/buritos/vlingo-wire/blob/master/src/main/java/io/vlingo/wire/fdx/bidirectional/rsocket/RSocketClientChannel.java#L73, but I am not sure about the implications of not making a copy and also if it is the same case in all of these methods I mentioned in this PR. What I don't understand is how copying the buffer addresses "payload might not be sent immediately". I can of-course investigate, but it would be faster is someone could explain the intent. Also, how can we be sure that the original buffer fits in the buffer that we get from the pool? The way I would remove the copying is by adding a constructor in `ConsumerByteBuffer` creates an instance from an existing `ByteBuffer`, instead of allocating a new one. Let me know if you remember anything that would help me understand this code faster than reading it and what's your thoughts about my assumptions so far.

Reply:
@buritos I like your idea of reusing a buffer, but I don't know if the copy-to buffer is owned by RSocket or vlingo. @alexguzun implemented the RSocket components so he can explain why this is done that way. He is in Lisbon and you can ping him on our team Slack workspace.

Reply:
Hi! The copy-to buffer ar owned by wire. They are required because unfortunatelly the rsocket and wire threads are separate meaning that a wire outbound might try to send the second message while the RSocket is still sending the first one. If we use the same buffer, the second message will overrite the first one.

Reply:
Thank you for explaining @alexguzun. I will spend some more time on this now that I clearly understand your motivation for copying. We might be able to get around this by pushing the acquisition of the buffers all the way up, reusing the buffers from the pool instead of holding on to particular instances. Does that make sense? Any thoughts?